src/libcollections/str.rs

```diff
@@ -164,17 +164,27 @@ impl ToOwned for str {
     }
 }
 
-/// Any string that can be represented as a slice.
+/// Methods for string slices.
 #[lang = "str"]
 #[cfg(not(test))]
 impl str {
-    /// Returns the length of `self` in bytes.
+    /// Returns the length of `self`.
+    ///
+    /// This length is in bytes, not [`char`]s, or graphemes. In other words,
+    /// it may not be what a human considers the length of the string.
```

Comment:
“may” is a bit weak IMO, since it will “not be what a human considers the length of the string” most of the time.